### PR TITLE
[Semi-Modular] Ports SEVA suit from oldbase.

### PR DESCRIPTION
**Pull request by Swiftfeather**
## About The Pull Request

Nanotrasen's mining division would like to (re)introduce our latest and greatest product! The SEVA suit! With this excellent new suit miners will be able to mine for extended periods, never needing to worry about things of the past like ash storms! And if they are the clumsy type even being on fire won't hurt them! This is proven to decrease miner fatality rates due to burning and storms!

5bca99c20d840b842d53f288c09fb28a

- The miners now start with one additional voucher in their bag, a suit voucher, this voucher can be redeemed at mining vendors for either SEVA suits or Explorer suits, this allows incoming miners to get right to working as soon as they arrive!

- The suits support digi, muzzled, and normal sprites.

- New vouchers are also purchasable from the mining vendor for 2,000 points.

- Please note, SEVA suits are completely unarmored against the creatures you may face while mining, and cannot be upgraded with goliath plates, making them unideal for combat applications.

- They are also too big to be placed in bags, so miners must choose between explorer and SEVA.

## Why It's Good For The Game

More mining suits is nice, also this sets the framework for additional suit voucher suits, perhaps even the explorer suit can be added so we always have them in stock for miners.

This suit is very much geared towards miners who are just mining, not those who are hunting.
It also allows miners to mine the whole time! No storm can stop them!

## Changelog
:cl:
add: Added suit vouchers, miners start with these and can redeem them for explorer suits or SEVA suits.
add: Added SEVA suit, purchasable with suit vouchers.
add: Suit  vouchers are purchasable from mining vendor for 2,000 points.
/:cl:

### CODE REVIEW

**Comment by GoldenAlpharex** [comment]
I like this, but I also wonder how it's going to feel like.

**Comment by YakumoChen** [comment]
> This suit is very much geared towards miners who are just mining, not those who are hunting.
It also allows miners to mine the whole time! No storm can stop them!

As someone who has spent nearly as much time as a Miner as they have in Security, I can DEFINITIVELY tell you that the SEVA suit is absolutely the _best_ suit for hunting if you know what you're doing. This in no way is a suit best suited for passive mining - it's the powergamer's choice.

SEVA's weaknesses (low armor, <s>weakness to goliath grabs</s>) mean extremely little. Good miners don't get caught, and don't take many hits. 

To compound this, the upside to SEVA is HUGE. It makes you fireproof, which makes the Drake mostly trivial to fight (being lit on fire is the biggest threat of Ash Drake and the regular explorer's suit only provides firesuit resistance), and lets you continue a fight if a storm happens to start during it, which in turn makes it extremely easy to get Drake Armor, and Drake Armor is more or less just a straight upgrade of SEVA.

EDIT: I should point out you have not ported the Goliath Grab weakness, making this all the more of an obvious choice to good miners.

File: modular_skyrat/modules/SEVA_suit/seva_obj.dm [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	resistance_flags = FIRE_PROOF
+
+/obj/item/clothing/head/hooded/explorer/seva
+	name = "SEVA Hood"
+	desc = "A fire-proof hood for exploring hot environments. Its design doesn't allow for upgrading with goliath plates."
+	icon = 'modular_skyrat/modules/SEVA_suit/icons/seva_hat.dmi'
+	worn_icon = 'modular_skyrat/modules/SEVA_suit/icons/equipped/seva_head.dmi'
+	icon_state = "seva"
+	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	armor = list("melee" = 10, "bullet" = 10, "laser" = 10, "energy" = 10, "bomb" = 35, "bio" = 50, "rad" = 25, "fire" = 100, "acid" = 25, "wound" = 5)
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/SEVA_suit/seva_obj.dm]:
```suggestion
	armor = list(MELEE = 10, BULLET = 10, LASER = 10, ENERGY = 10, BOMB = 35, BIO = 50, RAD = 25, FIRE= 100, ACID = 25, WOUND = 5)
```

File: modular_skyrat/modules/SEVA_suit/seva_obj.dm [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	desc = "A fire-proof suit for exploring hot environments. Its design doesn't allow for upgrading with goliath plates."
+	icon = 'modular_skyrat/modules/SEVA_suit/icons/seva_suit.dmi'
+	worn_icon = 'modular_skyrat/modules/SEVA_suit/icons/equipped/seva_suit.dmi'
+	worn_icon_digi = 'modular_skyrat/modules/SEVA_suit/icons/equipped/seva_suit_digi.dmi'
+	icon_state = "seva"
+	w_class = WEIGHT_CLASS_BULKY
+	max_heat_protection_temperature = FIRE_IMMUNITY_MAX_TEMP_PROTECT
+	heat_protection = CHEST|GROIN|LEGS|FEET|ARMS|HANDS
+	hoodtype = /obj/item/clothing/head/hooded/explorer/seva
+	armor = list("melee" = 15, "bullet" = 10, "laser" = 10, "energy" = 10, "bomb" = 35, "bio" = 50, "rad" = 25, "fire" = 100, "acid" = 25, "wound" = 5)
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/SEVA_suit/seva_obj.dm]:
```suggestion
	armor = list(MELEE = 15, BULLET = 10, LASER = 10, ENERGY = 10, BOMB = 35, BIO = 50, RAD = 25, FIRE = 100, ACID = 25, WOUND = 5)
```

File: modular_skyrat/modules/SEVA_suit/suit_voucher.dm [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	desc = "A token to redeem a new suit. Use it on a mining equipment vendor."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "mining_voucher"
+	w_class = WEIGHT_CLASS_TINY
+
+
+//Code to redeem new items at the mining vendor using the suit voucher
+//More items can be added in the lists and in the if statement.
+/obj/machinery/mineral/equipment_vendor/proc/RedeemSVoucher(obj/item/suit_voucher/voucher, mob/redeemer)
+	var/items = list("SEVA suit" = image(icon = 'modular_skyrat/modules/SEVA_suit/icons/seva_suit.dmi', icon_state = "seva"))
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/SEVA_suit/suit_voucher.dm]: Seems just a tad redundant for only having one suit. But it does mean we can add more in future I suppose.

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
Redundant in cold environments, it'll do.

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**